exporter/cat1_view_helper.go

```diff
@@ -140,11 +177,31 @@ func valueOrDefault(val interface{}, def interface{}) interface{} {
 	return def
 }
 
-// conditional assignment. returns the second value only if the first value is zero
-// TODO: make arguments and return type interface{}. add "value is empty, zero, or nil" to description
-func condAssign(first int64, second int64) int64 {
-	if first != 0 {
-		return first
+// conditional assignment. returns the second value only if the first value is empty, zero, or nil
+func condAssign(first interface{}, second interface{}) interface{} {
+	switch val := first.(type) {
```

Comment:
Ah sorry my mistake. Although you probably ought to change `val` to `first`, (so it's `switch first := first.(type)`) as per the example and the convention described there in the doc:
> It's also idiomatic to reuse the name in such cases, in effect declaring a new variable with the same name but a different type in each case.